silq/pulses/pulse_modules.py

```diff
@@ -231,44 +257,102 @@ def __init__(self,
                                              set_cmd=None,
                                              vals=vals.Bool())
 
+        self.t_start = Parameter(unit='s', set_cmd=None, initial_value=0)
         self.duration = Parameter(unit='s', set_cmd=None)
+        self.parameters['duration']._duration = None
+        self.t_stop = Parameter(unit='s', set_cmd=None)
+
         self.final_delay = Parameter(unit='s', set_cmd=None, vals=vals.Numbers())
         if final_delay is not None:
             self.final_delay = final_delay
         else:
             self.final_delay = self.default_final_delay
 
-        self.t_list = Parameter(initial_value=[0])
-        self.t_start_list = Parameter(initial_value=[])
-        self.t_stop_list = Parameter()
-
-        self.enabled_pulses = Parameter(initial_value=[], set_cmd=None,
-                                        vals=vals.Lists())
-        self.disabled_pulses = Parameter(initial_value=[], set_cmd=None,
-                                         vals=vals.Lists())
-        self.pulses = Parameter(initial_value=[], vals=vals.Lists(),
-                                set_cmd=None)
+        self.t_list = Parameter(initial_value=[0], snapshot_value=False)
+        self.t_start_list = Parameter(initial_value=[], snapshot_value=False)
+        self.t_stop_list = Parameter(snapshot_value=False)
+
+        self.pulse_sequences = Parameter(vals=vals.Iterables(), initial_value=())
+
+        self.my_enabled_pulses = Parameter(
```

Comment:
not a big deal, but "my_enabled_pulses" and "enabled_pulses" sound a bit not intuitive...

Reply:
Basically `enabled_pulses` includes all pulses from nested pulse sequences, while `my_enabled_pulses` is the ones that actually belong to this pulse sequence, and not a nested pulse sequence.
I'm happy to consider any other naming suggestions, otherwise I suggest we keep it this way.

Reply:
I see, ok, I was thinking something like "enabled_nested_pulses", but as it is can work as well